code/bngblaster/src/bbl_protocols.c

```diff
@@ -2202,12 +2202,40 @@ decode_icmpv6(uint8_t *buf, uint16_t len,
             }
             break;
         case IPV6_ICMPV6_NEIGHBOR_SOLICITATION:
+            if(len < 20) {
+                return DECODE_ERROR;
+            }
+            BUMP_BUFFER(buf, len, sizeof(uint32_t)); /* flags / reserved */
+            memcpy(&icmpv6->prefix.address, buf, IPV6_ADDR_LEN);
+            break;
         case IPV6_ICMPV6_NEIGHBOR_ADVERTISEMENT:
             if(len < 20) {
                 return DECODE_ERROR;
             }
             BUMP_BUFFER(buf, len, sizeof(uint32_t)); /* flags / reserved */
             memcpy(&icmpv6->prefix.address, buf, IPV6_ADDR_LEN);
+            BUMP_BUFFER(buf, len, IPV6_ADDR_LEN);
+            while(len >= 2) {
+                option = *buf;
+                BUMP_BUFFER(buf, len, sizeof(uint8_t));
+                option_len = (*buf) * 8;
+                BUMP_BUFFER(buf, len, sizeof(uint8_t));
+                if(option_len < 2 || len + 2 < option_len) {
```

Comment:
replace with `if(option_len < 2 || len < option_len-2) {` just to be similar to the code before (IPV6_ICMPV6_ROUTER_ADVERTISEMENT)

Reply:
done